Agrega tarea 5

hernan-peralta opened the pull request.

hernan-peralta committed:
Agrega tarea 5

fsodano commented on tareas/clase-5/index-mio.html:
```diff
@@ -0,0 +1,19 @@
+
+    <script src="tarea-clase-5-mio.js"></script>
+</body>
+</html>
```
falta nueva línea, eso es el simbolito que te aparece arriba

fsodano commented on tareas/clase-5/index-mio.html:
```diff
@@ -0,0 +1,19 @@
+</head>
+<body>
+    <form>
+        <input type="text" id=salarioanual placeholder="Ingrese su salario anual" autofocus="autofocus"/>
```
te faltan las `"` y se escribe en `kebab-case`. `id="salario-anual"`

fsodano commented on tareas/clase-5/index-mio.html:
```diff
@@ -0,0 +1,19 @@
+<body>
+    <form>
+        <input type="text" id=salarioanual placeholder="Ingrese su salario anual" autofocus="autofocus"/>
+        <button type="button" id=botoncalcular>Calcular</button>
```
```suggestion
        <button type="button" id="boton-calcular">Calcular</button>
```

fsodano commented on tareas/clase-5/index-mio.html:
```diff
@@ -0,0 +1,19 @@
+    <form>
+        <input type="text" id=salarioanual placeholder="Ingrese su salario anual" autofocus="autofocus"/>
+        <button type="button" id=botoncalcular>Calcular</button>
+        <input type="text" disabled id="salariomensual"/>
```
```suggestion
        <input type="text" disabled id="salario-mensual"/>
```

fsodano commented on tareas/clase-5/index3.html:
```diff
@@ -0,0 +1,25 @@
+        </form>
+    </div>
+    <br>
+    <button id=agregarform>Agregar nuevo video</button>
```
`agregarform` no es un nombre idóneo. Capaz `agregar-video` ya que el botón agrega... videos 😄

fsodano commented on tareas/clase-5/index3.html:
```diff
@@ -0,0 +1,25 @@
+            <b class="nombreclase">Clase 1</b>
+            <input type="number" class=horas placeholder="Duración en horas" autofocus="autofocus"/>
+            <input type="number" class=minutos placeholder="Duración en minutos"/>
+            <input type="number" class=segundos placeholder="Duración en segundos"/>
```
Podrías arrancar sin videos

fsodano commented on tareas/clase-5/index3.js:
```diff
@@ -0,0 +1,31 @@
+
+agregarClase.onclick = () => {
+    cantidadClases ++;
+    document.querySelector("div").appendChild(clonDeFormulario.cloneNode(true));
```
acá estás clonando al clon del formulario

hernan-peralta replied:
Es que al principio clono el formulario vacío y luego clono ese clon (que no uso), porque si clono uno que ya tenga valores me los clona también. Una solución media sucia quizá.

fsodano commented on tareas/clase-5/index3.js:
```diff
@@ -0,0 +1,31 @@
+agregarClase.onclick = () => {
+    cantidadClases ++;
+    document.querySelector("div").appendChild(clonDeFormulario.cloneNode(true));
+    document.getElementsByClassName("nombreclase")[document.getElementsByClassName("nombreclase").length-1].innerHTML = "Clase " + cantidadClases; // en el lado izquierdo de la asignacion selecciono el ultimo elemento del array que contiene la clase "nombreclase"
```
Podés usar el pseudo-selector `last-of-type` para traer el último elemento en vez de traer todos y sólo elegir el último.

```suggestion
    document.querySelector("nombreclase:last-of-type").innerHTML = "Clase " + cantidadClases;
```

fsodano replied:
De hecho, esto no funciona. Funciona sólo si usás el nombre del tag, mala mía. https://css-tricks.com/almanac/selectors/l/last-of-type/

hernan-peralta replied:
¡Gracias por el aviso! ahí lo cambio.

fsodano commented on tareas/clase-5/index3.js:
```diff
@@ -0,0 +1,31 @@
+    let segundos = 0;
+    for (let i =0; i<document.getElementsByClassName("segundos").length; i++){
+        segundos += Number(document.getElementsByClassName("segundos")[i].value);
+    }
```
En vez de hacer 2 veces `getElementsByClassaName("segundos")`, usalo una única vez y guardalo en una variable. Lo mismo para los posteriores,

fsodano commented on tareas/clase-5/index4.js:
```diff
@@ -0,0 +1,83 @@
+let numeros = [];
+
+for (let i =0; i<$lista.length; i++){
+    numeros.push((Number($lista[i].innerHTML)));
```
Tenés unos paréntesis de más

```suggestion
    numeros.push(Number($lista[i].innerHTML));
```

hernan-peralta committed:
Update tareas/clase-5/index-mio.html

Co-Authored-By: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

hernan-peralta committed:
Update tareas/clase-5/index-mio.html

Co-Authored-By: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

hernan-peralta committed:
Update tareas/clase-5/index4.js

Co-Authored-By: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

hernan-peralta committed:
Aplica correcciones sugeridas a tarea-clase-5

hernan-peralta committed:
Corrije nombre id de boton por otro mas idoneo

hernan-peralta committed:
Merge branch 'tarea-clase-5' of https://github.com/hernan-peralta/introduccion-a-js into tarea-clase-5

hernan-peralta committed:
Mejoras en la legibilidad del codigo

Arreglada tambien la funcionalidad de ejercicio 3 en lo referente a la cuenta de videos agregados.
